### PR TITLE
Fix Metricbeat/vsphere - Error "datastore '*' not found" #4879

### LINKED ISSUE #4879: [Metricbeat][Vsphere] - Error "datastore '*' not found"  #WIP
When collecting from a vsphere 6.5, finder fails with a datastore (or host) '*' not found error.

### PR DESCRIPTION
**Pull request by amandahla**
We needed to change the way we use govmomi for collect metrics and now it works with versions 6.0/6.5. 

We have to remove 'datacenter' field also. 

Config.yml was corrected.

### CODE REVIEW

**Comment by exekias** [comment]
Could you please add a CHANGELOG entry?

Also, do you know if the new client version may help on this? https://github.com/elastic/beats/issues/4673, have you experienced any issue like that?

**Comment by tsg** [comment]
jenkins, test it

**Comment by amandahla** [comment]
@exekias  I updated the govmomi (0.15.0 now) and changed the way to collect. For the issue #4673 , I think that is possible to use something like this https://github.com/vmware/govmomi/issues/224 
I'm not sure how it will work :-|

**Comment by exekias** [comment]
ok to test.

@amandahla is this ready for a new review?

**Comment by amandahla** [comment]
@exekias Yes. Just had that issue with the flag 'version'.

**Comment by exekias** [comment]
code looks good to me, there are some issues building it for other systems, do you have any idea of what's going on there? https://travis-ci.org/elastic/beats/jobs/265294456

**Comment by exekias** [comment]
I'll have a look into the issue

File: metricbeat/module/vsphere/host/host.go [hunk 1]
```diff
@@ -48,87 +45,78 @@ func New(base mb.BaseMetricSet) (mb.MetricSet, error) {
...
+
+	// Retrieve summary property for all hosts.
+	var hst []mo.HostSystem
+	err = v.Retrieve(ctx, []string{"HostSystem"}, []string{"summary"}, &hst)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, hs := range hst {
+		totalCpu := int64(hs.Summary.Hardware.CpuMhz) * int64(hs.Summary.Hardware.NumCpuCores)
```
[the remark this answers is not in the document]
Comment (amandahla) [reply metricbeat/module/vsphere/host/host.go]: Solved.

File: metricbeat/module/vsphere/host/host.go [hunk 2]
```diff
@@ -48,87 +45,78 @@ func New(base mb.BaseMetricSet) (mb.MetricSet, error) {
...
+		freeCpu := int64(totalCpu) - int64(hs.Summary.QuickStats.OverallCpuUsage)
```
[the remark this answers is not in the document]
Comment (amandahla) [reply metricbeat/module/vsphere/host/host.go]: Solved.

**Comment by ruflin** [comment]
@amandahla @exekias What is the status here? 

@amandahla Any chance you could rebase this on master and then we could have a look again.

**Comment by exekias** [comment]
I had a look at the issue but couldn't get it working, looks like the client lacks support for many architectures  :/

**Comment by amandahla** [comment]
Using session we were getting errors with architectures but now opening a new client on every fetch seems ok. It's the same way like before but instead doing a new connection only once, we do it for every fetch.

**Comment by ruflin** [comment]
@amandahla Could you rebase this one again for the changelog?

**Comment by ruflin** [comment]
@amandahla I did 2 commits myself to see if I can fix the CHANGELOG issue and getting this in.

**Comment by amandahla** [comment]
@ruflin Thanks!

**Comment by exekias** [comment]
Awesome contribution @amandahla, I'm glad this got finally in!! :tada: :tada:

**Comment by amandahla** [comment]
@exekias Thank you, I'm very happy with that! :smile:

**Pull request merged**